src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashAditionalParameterEnvironmentContributor.java

```diff
@@ -10,40 +10,32 @@
 
 @Extension
 public class StashAditionalParameterEnvironmentContributor extends EnvironmentContributor {
-    private static Set<String> params =
-            new HashSet<String>(Arrays.asList("sourceBranch",
-                    "targetBranch",
-                    "sourceRepositoryOwner",
-                    "sourceRepositoryName",
-                    "pullRequestId",
-                    "destinationRepositoryOwner",
-                    "destinationRepositoryName",
-                    "pullRequestTitle",
-                    "sourceCommitHash",
-                    "destinationCommitHash"));
-
     @Override
     public void buildEnvironmentFor(@Nonnull Run r, @Nonnull EnvVars envs, @Nonnull TaskListener listener) throws IOException, InterruptedException {
-        StashCause cause = (StashCause) r.getCause(StashCause.class);
-        if (cause == null) {
-            return;
-        }
-        ParametersAction pa = r.getAction(ParametersAction.class);
-        for (String param : params) {
-            addParameter(param, pa, envs);
+        if (r instanceof AbstractBuild) {
+            AbstractBuild build = (AbstractBuild) r;
+            AbstractBuild rootBuild = build.getRootBuild();
+
+            StashCause cause = (StashCause) rootBuild.getCause(StashCause.class);
+            if (cause != null) {
```

Comment:
minor: can we leave the `cause == null` check at the top, just to make the method simpler?

Reply:
nvm: I just noticed this is cause from rootBuild and not from the run

Reply:
I don't want to call `super.buildEnvironmentFor()` conditionally. It's actually an empty function in `EnvironmentContributor`, but I want to be consistent. Either we call it and don't return prematurely, or we leave it out completely.